cmd/ipfs/daemon.go

```diff
@@ -789,9 +787,8 @@ func serveHTTPGateway(req *cmds.Request, cctx *oldcmds.Context) (<-chan error, e
 		return nil, fmt.Errorf("serveHTTPGateway: GetConfig() failed: %s", err)
 	}
 
-	writable, writableOptionFound := req.Options[writableKwd].(bool)
-	if !writableOptionFound {
-		writable = cfg.Gateway.Writable
+	if cfg.Gateway.Writable {
+		return nil, fmt.Errorf("serveHTTPGateway: writable gateways are no longer supported, please remove .Gateway.Writable from your configuration file")
```

Comment:
I filled https://github.com/ipfs/kubo/issues/9622 – if we are removing this, then we also need to turn `Gateway.Writable` to be an optional `Flag`, so it does not pollute config. 

But maybe we could keep it for 0.19, so people have time to migrate to something else? Suddenly removing feature that has been around for 5-8 years is  not cool, and we should try to avoid breaking  if we can (like we did with `ipfs object` API).